pkg/rhcos/cache/cache.go

```diff
@@ -275,6 +275,9 @@ func DownloadImageFile(baseURL string, applicationName string) (string, error) {
 		u.uncompressedSHA256 = uncompressedSHA256[0]
 		q.Del("sha256")
 		parsedURL.RawQuery = q.Encode()
+	} else if sha256Checksum != "" {
```

Comment:
If passed, shouldn't this field have the priority (and equal to `q["sha256"]` if present?) I was wondering if there could be the case of a different value

Reply:
I think they should be the same, will add a check.